crates/wash-cli/src/bin/wash.rs

```diff
@@ -355,3 +442,40 @@ async fn main() {
 fn experimental_error_message(command: &str) -> anyhow::Result<CommandOutput> {
     bail!("The `wash {command}` command is experimental and may change in future releases. Set the `WASH_EXPERIMENTAL` environment variable or `--experimental` flag to `true` to use this command.")
 }
+
+/// Helper for loading plugins. This function does not return an error because it will exit the
+/// process in case of a fatal error. As such, this shouldn't be called after the CLI is parsed.
+async fn load_plugins() -> (SubcommandRunner, PathBuf) {
+    // We need to use env vars here because the plugin loading needs to be initialized before
+    // the CLI is parsed
+    let plugin_dir = match ensure_plugin_dir(std::env::var("WASH_PLUGIN_DIR").ok()).await {
+        Ok(dir) => dir,
+        Err(e) => {
+            eprintln!("Could not load wash plugin directory: {}", e);
+            std::process::exit(1);
```

Comment:
what do you think about not exiting here and just letting `wash` run without plugins?
I would imagine to be able to run e.g. `wash up` even if plugin directory failed to get initialized (because my file system is read-only for example)

Reply:
Good point, I'll change it out to just write an error